scapy/contrib/coap_socket.py

```diff
@@ -0,0 +1,1001 @@
+# SPDX-License-Identifier: GPL-2.0-only
+# This file is part of Scapy
+# See https://scapy.net/ for more information
+# Copyright (C) 2024 eHonnef <contact@honnef.net>
+
+# scapy.contrib.description = CoAP Socket Library / RFC 7252
+# scapy.contrib.status = library
+
+import logging
+import random
+import socket
+import time
+
+# Typing imports
+from typing import (
+    Optional,
+    Union,
+    Tuple,
+    Any,
+    cast,
+    Type
+)
+
+from scapy.error import Scapy_Exception
+from scapy.packet import Packet
+from scapy.contrib.coap import CoAP, coap_options, coap_codes
+from scapy.contrib.isotp.isotp_soft_socket import TimeoutScheduler
+from scapy.data import MTU
+from scapy.utils import EDecimal
+from scapy.automaton import ObjectPipe, select_objects
+
+from scapy.supersocket import SuperSocket, SimpleSocket
+
+"""
+CoAP message request codes (RFC 7252 @ section-5.8.1)
+"""
+EMPTY_MESSAGE = 0
+GET = 1
+POST = 2
+PUT = 3
+DELETE = 4
+COAP_REQ_CODES = [GET, POST, PUT, DELETE]
+"""
+CoAP message response codes (RFC 7252 @ section-12.1.2)
+Also, from scapy.contrib.coap.coap_codes
+"""
+EMPTY_ACK = EMPTY_MESSAGE
+CONTENT_205 = 69
+NOT_FOUND_404 = 132
+NOT_ALLOWED_405 = 133
+NOT_IMPLEMENTED_501 = 161
+"""
+CoAP content type (RFC 7252 @ section-12.3)
+"""
+CF_TEXT_PLAIN = b"\x00"
+CF_APP_LINK_FORMAT = b"\x28"
+CF_APP_XML = b"\x29"
+CF_APP_OCTET_STREAM = b"\x2A"
+CF_APP_EXI = b"\x2F"
+CF_APP_JSON = b"\x32"
+"""
+CoAP options (RFC 7252 @ section-5.10)
+"""
+PAYMARK = b"\xff"
+URI_PATH = 11
+CONTENT_FORMAT = 12
+"""
+CoAP message type
+"""
+CON = 0
+NON = 1
+ACK = 2
```

Comment:
In general, I think these definitions should be part of coap.py

Reply:
Moved